components/mpas-ocean/src/shared/mpas_ocn_scaled_sfwf.F

```diff
@@ -0,0 +1,557 @@
+! Copyright (c) 2013,  Pacific Northwest National Laboratory, (PNNL)
+! and the University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at http://mpas-dev.github.io/license.html
+!
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  ocn_scaled_sfwf
+!> \brief MPAS ocean scale data freshwater fluxes
+!> \author Shixuan Zhang 
+!> \date   May 2025
+!> \details
+!>  Computes a precipitation factor to multiply the fresh water flux
+!>  due to precipitation uniformly to ensure a balance of fresh water
+!>  at the ocean surface.
+!>  Designed following precip_adjustment subroutine located at:
+!>    https://github.com/ESCOMP/POP2-CESM/blob/master/source/forcing_sfwf.F90
+!
+!-----------------------------------------------------------------------
+
+module ocn_scaled_sfwf
+
+    use mpas_kind_types
+    use mpas_derived_types
+    use mpas_global_sum_mod
+    use mpas_timekeeping
+    use mpas_timer
+
+    use ocn_config
+    use ocn_mesh
+
+    use shr_kind_mod, only: & 
+        SHR_KIND_R8
+
+    use shr_const_mod
```

Comment:
```suggestion
```
I would prefer to see the SHR constants you need added to ocn_constants as it is done for `pi`

Reply:
@zhangshixuan1987 is likely following my bad example :-(
https://github.com/E3SM-Project/E3SM/blob/7ac7d34707ca22c2555ca86a79a5c07b5a54fc4d/components/mpas-ocean/src/shared/mpas_ocn_scaled_dismf.F#L33